### PR TITLE
change okHttpClient to Retrofit

### PR DESCRIPTION
**Pull request by deyasix**
I deleted class Client that uses OkHttpClient for uploading data. I added Retrofit instead of OkHttpClient. I also added appropriate data classes for getting response. I created interface APIService, class StatisticService and object RetrofitClient. Data class DayStatistic also was changed.

### CODE REVIEW

File: app/src/main/java/com/fivesysdev/standwithukraine/ui/StatisticFragment.kt [hunk 1]
```diff
@@ -62,7 +61,7 @@ class StatisticFragment : Fragment(R.layout.fragment_statistic) {
     private fun setDayStatistic(dayStatistic: DayStatistic?) {
         checkBlockNextButton()
         if (dayStatistic != null) {
-            binding.recyclerview.adapter = StatisticAdapter(dayStatistic.statistic)
+            binding.recyclerview.adapter = StatisticAdapter(dayStatistic.getStatisticsPair())
```
Comment (okorchysta) [inline app/src/main/java/com/fivesysdev/standwithukraine/ui/StatisticFragment.kt]: Why do you create a new adapter each time the statistics change?

**Review by okorchysta — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/fivesysdev/standwithukraine/data/DayStatisticRepositoryImpl.kt [hunk 2]
```diff
@@ -7,16 +7,15 @@ import java.util.concurrent.FutureTask
...
     }
 
     private fun updateStatistics() {
         val task = Callable {
-            val client = Client()
-            return@Callable client.getStatisticsFromDate(fromDate.toString())
+            return@Callable StatisticService().getStatisticFromDate(fromDate.toString())
         }
         val future = FutureTask(task)
         Thread(future).start()
```
Comment (okorchysta) [inline app/src/main/java/com/fivesysdev/standwithukraine/data/DayStatisticRepositoryImpl.kt]: I've just noticed the `statistics.stream()` in `addAll(list: List<DayStatistic>) ` I'm sure you can find a better way to do this:)

**Review by okorchysta — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/fivesysdev/standwithukraine/ui/StatisticFragment.kt (hunk 1 above)
Comment (deyasix) [reply app/src/main/java/com/fivesysdev/standwithukraine/ui/StatisticFragment.kt]: I need somehow change the list of elements. So I was thinking about creating a new adapter each time or adding a new method to the adapter class, that replaces data in the list. The first option seemed the simplest, but not the smartest I guess.

Comment (okorchysta) [reply app/src/main/java/com/fivesysdev/standwithukraine/ui/StatisticFragment.kt]: there are a bunch of methods like notifyDataSetChanged for this purpose.
Please, check the documentation/articles on how to properly change data in list

File: app/src/main/java/com/fivesysdev/standwithukraine/data/DayStatisticRepositoryImpl.kt (hunk 2 above)
Comment (deyasix) [reply app/src/main/java/com/fivesysdev/standwithukraine/data/DayStatisticRepositoryImpl.kt]: Thanks, I'll rewrite this part of the code.

File: app/src/main/java/com/fivesysdev/standwithukraine/ui/StatisticFragment.kt (hunk 1 above)
Comment (deyasix) [reply app/src/main/java/com/fivesysdev/standwithukraine/ui/StatisticFragment.kt]: Okay, I'll check

File: app/src/main/java/com/fivesysdev/standwithukraine/data/DayStatisticRepositoryImpl.kt [hunk 3]
```diff
@@ -2,28 +2,17 @@ package com.fivesysdev.standwithukraine.data
...
-    private fun updateStatistics() {
-        val task = Callable {
-            val client = Client()
-            return@Callable client.getStatisticsFromDate(fromDate.toString())
-        }
-        val future = FutureTask(task)
-        Thread(future).start()
-        addAll(future.get())
+    private suspend fun updateStatistics() {
+        addAll(StatisticService().getStatisticFromDate(fromDate.toString()))
```
Comment (okorchysta) [inline app/src/main/java/com/fivesysdev/standwithukraine/data/DayStatisticRepositoryImpl.kt]: why do you create a StatisticService every time here?

File: app/src/main/java/com/fivesysdev/standwithukraine/ui/StatisticFragment.kt [hunk 4]
```diff
@@ -36,18 +36,27 @@ class StatisticFragment : Fragment(R.layout.fragment_statistic) {
...
         setListeners()
         setRecyclerView()
-        setDayStatistic(viewModel.getCurrentDayStatistic())
+        observeState()
+    }
+
+    private fun observeState() {
+        viewModel.loading.observe(viewLifecycleOwner) {
+            if (it) {
+                binding.emptydata.root.visibility = View.GONE
```
Comment (okorchysta) [inline app/src/main/java/com/fivesysdev/standwithukraine/ui/StatisticFragment.kt]: by the way, it's very comfortable to have smth like this:

```
fun View.setVisibility(isVisible: Boolean) {
    this.visibility = if (isVisible) View.VISIBLE else View.GONE
}

fun View.setSoftVisibility(isVisible: Boolean) {
    this.visibility = if (isVisible) View.VISIBLE else View.INVISIBLE
}

```

File: app/src/main/java/com/fivesysdev/standwithukraine/ui/StatisticFragment.kt [hunk 5]
```diff
@@ -61,11 +70,14 @@ class StatisticFragment : Fragment(R.layout.fragment_statistic) {
...
-            binding.recyclerview.adapter = StatisticAdapter(dayStatistic.statistic)
-        }
+        updateDataList(dayStatistic)
         binding.tvDate.text = viewModel.date.toString()
-        setEmptyDataObserver()
+    }
+
+    private fun updateDataList(dayStatistic: DayStatistic?) {
+        data.clear()
+        data.addAll(dayStatistic?.getStatisticsPair()?:listOf())
```
Comment (okorchysta) [inline app/src/main/java/com/fivesysdev/standwithukraine/ui/StatisticFragment.kt]: please, use code formatting tool
To automatically format your code in the current source code window, use Cmd+Option+L (on Mac) or Ctrl+Alt+L (on Windows and Linux).

**Review by okorchysta — CHANGES_REQUESTED** [review]

**Review by okorchysta — APPROVED** [review]

**Pull request merged**